src/twisted/conch/scripts/ckeygen.py

```diff
@@ -202,10 +203,18 @@ def _defaultPrivateKeySubtype(keyType):
         return "PEM"
 
 
-def printFingerprint(options):
+def getKeyOrDefault(options):
     if not options["filename"]:
         filename = os.path.expanduser("~/.ssh/id_rsa")
-        options["filename"] = input("Enter file in which the key is (%s): " % filename)
+        if platform.system() == "Windows":
+            filename = os.path.expandvars(R"%HOMEPATH %\.ssh\id_rsa")
+        options["filename"] = (
+            input("Enter file in which the key is (%s): " % filename) or filename
+        )
+
+
+def printFingerprint(options):
+    getKeyOrDefault(options)
```

Comment:
this is kind of auto-magic code.

Maybe this will make it easier to read and understand what it does

```suggestion
    options["filename"] = resolveFilename(options)
```

Reply:
See above, I've added a docstring.